src/System.Net.Http/src/System/Net/Http/Headers/KnownHeader.cs

```diff
@@ -12,34 +13,36 @@ internal sealed class KnownHeader
         private readonly HttpHeaderType _headerType;
         private readonly HttpHeaderParser _parser;
         private readonly string[] _knownValues;
+        private readonly byte[] _asciiBytesWithColonSpace;
 
-        public KnownHeader(string name, HttpHeaderType headerType, HttpHeaderParser parser, string[] knownValues = null)
+        public KnownHeader(string name) : this(name, HttpHeaderType.Custom, null)
         {
             Debug.Assert(!string.IsNullOrEmpty(name));
             Debug.Assert(HttpRuleParser.GetTokenLength(name, 0) == name.Length);
-            Debug.Assert(headerType != HttpHeaderType.Custom);
```

Comment:
We should retain the checks these asserts provide, i.e.:
Custom headers can't have a parser or known values
Non-custom headers must have a parser

Reply:
Ok.